File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        HashMap<String,Integer> map = new LinkedHashMap<>();
        
        for(int i=0; i< name.length; i++){
            map.put(name[i], yearning[i]); // 추억하는 사람 이름에 점수 저장
        }

        for(int i=0; i< photo.length; i++){
            String[] persons = photo[i]; // 그룹별 배열 생성
            int score = 0;
            
            for(int j=0; j<persons.length; j++){
            	
                String person = persons[j];
                if(map.containsKey(person)){ // map의 key에 photo 속 사람이 있다면
                    score+=map.get(person); // 점수 추가
                }
            }
            answer[i]=score;
        }
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    
    for(let i = 0; i < photo.length; i++){
        let result = 0;
        for(let j = 0; j < photo[i].length; j++){
            for(let k = 0; k < name.length; k++){
                if(name[k] === photo[i][j]) {
                    result += yearning[k];
                }
            }
        }
        answer.push(result);
    }
    return answer;
}